mlflow/transformers/llm_inference_utils.py

```diff
@@ -13,6 +19,129 @@
 }
 
 
+def infer_signature_from_llm_inference_task(
+    inference_task: str, signature: Optional[ModelSignature] = None
+) -> ModelSignature:
+    """
+    Sets the model signature if provided.
+    When a MLflow inference task is given, raise exception if signature is given,
+    otherwise infer the signature from the task.
+    """
+    if signature is not None:
+        if inference_task:
+            raise MlflowException(
+                f"When `task` is specified as `{inference_task}`, the signature would "
+                "be set by MLflow. Please do not set the signature."
+            )
+        return signature
+
+    if inference_task == _LLM_INFERENCE_TASK_CHAT:
+        signature = ModelSignature(
+            inputs=CHAT_MODEL_INPUT_SCHEMA,
```

Comment:
Can we also fix output schema, or it is more dynamic?

Reply:
Yes, now added `_LLM_INFERENCE_COMPLETIONS_OUTPUT_SCHEMA` and `_LLM_INFERENCE_CHAT_OUTPUT_SCHEMA`.